rxjava2-jdbc/src/main/java/org/davidmoten/rx/jdbc/Util.java

```diff
@@ -287,39 +287,43 @@ static void closeCallableStatementAndConnection(NamedCallableStatement stmt) {
         closePreparedStatementAndConnection(stmt.stmt);
     }
 
-    static NamedPreparedStatement prepare(Connection con, String sql) throws SQLException {
-        return prepare(con, 0, sql);
+    static NamedPreparedStatement prepare(Connection con, String sql, int queryTimeoutSec) throws SQLException {
+        return prepare(con, 0, sql, queryTimeoutSec);
     }
 
-    static NamedPreparedStatement prepare(Connection con, int fetchSize, String sql) throws SQLException {
+    static NamedPreparedStatement prepare(Connection con, int fetchSize, String sql, int queryTimeoutSec) throws SQLException {
         // TODO can we parse SqlInfo through because already calculated by
         // builder?
         SqlInfo info = SqlInfo.parse(sql);
         log.debug("preparing statement: {}", sql);
-        return prepare(con, fetchSize, info);
+        return prepare(con, fetchSize, info, queryTimeoutSec);
     }
 
-    static PreparedStatement prepare(Connection connection, int fetchSize, String sql, List<Parameter> parameters)
+    static PreparedStatement prepare(Connection connection, int fetchSize, String sql, List<Parameter> parameters, int queryTimeoutSec)
             throws SQLException {
         // should only get here when parameters contains a collection
         SqlInfo info = SqlInfo.parse(sql, parameters);
         log.debug("preparing statement: {}", info.sql());
-        return createPreparedStatement(connection, fetchSize, info);
+        return createPreparedStatement(connection, fetchSize, info, queryTimeoutSec);
     }
 
-    private static NamedPreparedStatement prepare(Connection con, int fetchSize, SqlInfo info) throws SQLException {
-        PreparedStatement ps = createPreparedStatement(con, fetchSize, info);
+    private static NamedPreparedStatement prepare(Connection con, int fetchSize, SqlInfo info, int queryTimeoutSec) throws SQLException {
+        PreparedStatement ps = createPreparedStatement(con, fetchSize, info, queryTimeoutSec);
         return new NamedPreparedStatement(ps, info.names());
     }
 
-    private static PreparedStatement createPreparedStatement(Connection con, int fetchSize, SqlInfo info)
+    private static PreparedStatement createPreparedStatement(Connection con, int fetchSize, SqlInfo info, int queryTimeoutSec)
             throws SQLException {
         PreparedStatement ps = null;
         try {
             ps = con.prepareStatement(info.sql(), ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
             if (fetchSize > 0) {
                 ps.setFetchSize(fetchSize);
             }
+
+            if (queryTimeoutSec > 0) {
```

Comment:
I'd leave the 0 value check out so that if a default query timeout is set in a connection string (possible with some drivers) then the user can set no timeout by query if they want.

Reply:
Sure, I have changed it in the recent commit